File: question/leetcode/leetcode0011.Container-With-Most-Water/leetcode0011.java
class Solution {
    public int maxArea(int[] height) {
        Integer left = 0, right = height.length - 1;
        Integer capacity = 0;
        while (left < right) {
            var cur = Math.min(height[left], height[right]) * (right - left);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
            if (cur > capacity) {
                capacity = cur;
            }
        }

        return capacity;
    }

}

File: question/leetcode/leetcode0011.Container-With-Most-Water/leetcode0011.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, riight = 0, len(height) - 1
        capacity = 0
        while left < riight:
            cur_capacity = min(height[left], height[riight]) * (riight - left)
            if height[left] < height[riight]:
                left += 1
            else:
                riight -= 1
            if cur_capacity > capacity:
                capacity = cur_capacity
        return capacity
